### PR TITLE
cosmos spark 4.2.0 release

### PR DESCRIPTION
**Pull request by moderakh**
cosmos spark 4.2.0 release

### CODE REVIEW

**Review by aayush3011 — APPROVED** [review]

File: sdk/cosmos/azure-cosmos-spark_3-1_2-12/CHANGELOG.md [hunk 1]
```diff
@@ -1,9 +1,16 @@
...
+### 4.2.0
 #### Configuration Changes
 * Changed the default value of `spark.cosmos.read.inferSchema.forceNullableProperties` from `false` to `true` based on user feedback, see [PR](https://github.com/Azure/azure-sdk-for-java/pull/22049).
 
+#### Key Bug Fixes
+* Fixes conversion for MapType schema, see [PR](https://github.com/Azure/azure-sdk-for-java/pull/22291).
+* Not-nullable properties to include "id", see [PR](https://github.com/Azure/azure-sdk-for-java/pull/22143).
+* Support CustomQuery to be used for inference, see [PR](https://github.com/Azure/azure-sdk-for-java/pull/22079).
+* Fixes collision resolution on schema inference, see [PR](https://github.com/Azure/azure-sdk-for-java/pull/21933).
+* Fixes max length of userAgent header, see [PR](https://github.com/Azure/azure-sdk-for-java/pull/22018).
```
Comment (FabianMeiswinkel) [inline sdk/cosmos/azure-cosmos-spark_3-1_2-12/CHANGELOG.md:12]: This should also include a note about determining the micro batch size dynamically now

Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos-spark_3-1_2-12/CHANGELOG.md:12]: * Improves bulk ingestion throttling rate by dynamically adjusting the max. micro-batch size

**Review by FabianMeiswinkel — APPROVED** [review]
Please add the proposed item to the changelog (or something along the lines)

File: sdk/cosmos/azure-cosmos-spark_3-1_2-12/CHANGELOG.md (hunk 1 above)
Comment (moderakh) [reply sdk/cosmos/azure-cosmos-spark_3-1_2-12/CHANGELOG.md:12]: done thanks for catching this.

**Pull request merged**